### DISCUSSION TITLE
How to incorporate ipc::channel with asio

### DISCUSSION THREAD

**Discussion by jbbjarnason** [mutouyun/cpp-ipc#99] [Q&A] [answered]
On Linux, is it possible to get a file descriptor for a ipc::channel to assign to asio::posix::stream_descriptor? So I could let asio handle epoll for that given file descriptor and await new events from the ipc::channel, such as receive.

For reference, https://think-async.com/Asio/asio-1.22.0/doc/asio/reference/posix__stream_descriptor.html

**Comment by mutouyun** [comment, marked as answer]
I will carefully consider this issue in the refactoring branch.
